src/openrct2/world/Park.cpp

```diff
@@ -117,6 +119,22 @@ namespace OpenRCT2::Park
 
     static uint32_t calculateSuggestedMaxGuests()
     {
+#ifdef ENABLE_SCRIPTING
+        auto& hookEngine = GetContext()->GetScriptEngine().GetHookEngine();
+        if (hookEngine.HasSubscriptions(HOOK_TYPE::PARK_CALCULATE_GUEST_CAP))
+        {
+            auto& gameState = GetGameState();
+            auto ctx = GetContext()->GetScriptEngine().GetContext();
+            auto obj = DukObject(ctx);
+            obj.Set("suggestedMaxGuests", gameState.SuggestedGuestMaximum);
+            auto e = obj.Take();
+            hookEngine.Call(HOOK_TYPE::PARK_CALCULATE_GUEST_CAP, e, true);
+
+            auto suggestedMaxGuests = AsOrDefault(
+                e["suggestedMaxGuests"], static_cast<int32_t>(gameState.SuggestedGuestMaximum));
+            return suggestedMaxGuests;
+        }
+#endif
```

Comment:
Shouldn't this happen after the guest cap has been recalculated? Now you're passing in the old guest cap from 13 seconds ago, which 13 seconds later is the value your plugin assigned it to. 😅

Reply:
I have tested it and it should not make a difference as it is returning a value. I placed it at the bottom and the hook stopped working.

Reply:
It should make a difference. There's two parts here:
a. the input to the hook, which is what a plugin can read when the suggested max changes.
b. the output of the hook, which is what a plugin can set to their desire.

The first one (a.) is the one that lags 13 seconds behind and is incorrect on later iterations, so the setup described [here](https://github.com/OpenRCT2/OpenRCT2/pull/21521#issuecomment-2016466947) wouldn't work.

You wouldn't necessarily need to return within this hook code, btw. It should become part of the calculation at the end, where plugins can either read the original outcome, or alter it to their desire with having an option of using the original outcome as part of their modification.

Reply:
Ah, I see what you mean. I will move the hook to the bottom of the function.